src/World.cpp

```diff
@@ -1037,14 +1064,122 @@ void cWorld::Tick(std::chrono::milliseconds a_Dt, std::chrono::milliseconds a_La
+	cRoot * Root = cRoot::Get();
+	size_t Ram = Root->GetApproximateChunkRAM();  // RAM used by all world's loaded chunks combined (approximated) in MiB
+	size_t MaxRam = Root->GetMaxRAMChunks();      // Maximum allowed RAM use of all worlds in MiB
+
+	/* LOGD("%s: GlobalRam: %zu, myChunks: %zu, Unloadable: %zu", GetName().c_str(), Ram, GetNumChunks(), GetMemoryCounter().GetCanUnloadCount()); */
+
+	// If the setting is set to minimize ram
+	if (Root->GetMinimizeRam())
 	{
-		UnloadUnusedChunks();
+		// Save all chunks if RAM is too much
+		if (Ram > MaxRam)
+		{
+			if (m_WorldAge - m_LastSave > std::chrono::seconds(20))
+			{
+				LOGD("###### %s: Saving all unused dirty chunks", GetName().c_str());
+				SaveChunks(true);
+			}
+		}
+
+		// Unload all unused non dirty chunks anyways
+		return CHUNK_CHANCE_ACCURACY;
+		// The chance of a chunk to unload the next tick is random, and is equal to
+		// <returned value> out of CHUNK_CHANCE_ACCURACY.
+		// returning CHUNK_CHANCE_ACCURACY means all unused non dirty chunks will
+		// unload with 100% certainty.
 	}
 
+	// If we're here, then the setting is not set to minimize RAM, therefore we
+	// take advantage of the RAM we have and use it as a chunk cache.
+	// We unload as lazily as possible.
+
+	// *** The code below explained briefly ***
+	// Keeping chunks in the RAM is a good thing and RAM acts as a chunk cache.
+	// We want to unload chunks as little as possible.
+	// If maxRam is exceeded, we unload just enough to go below the limit.
+	// If we really cannot stop RAM growth because too many chunks are dirty and unloadable,
+	// then we call the save cycle.
+
+	if (Ram > MaxRam)  // If all worlds combined should free something
+	{
+		// Calculate how much RAM all worlds combined should free (MiB)
+		size_t RamFreeGoal = Root->GetApproximateChunkRAM() - Root->GetMaxRAMChunks();
+
+		// Calculate how many chunks all worlds should free, assuming all chunks consume equal RAM.
+		size_t ChunkUnloadGoal = (RamFreeGoal * Root->GetTotalChunkCount()) / Ram;
+
+		// How many unloadable chunks all worlds have
+		size_t UnloadableChunks = Root->GetTotalCanUnloadCount();
+
+		// If the chunk unload goal is bigger than the amount of unloadable chunks the worlds have,
+		// then we consider saving chunks in the following if block, and we may unload all unused
+		// non dirty chunks we have. We are patient, and we don't save straight away.
+		if (ChunkUnloadGoal > UnloadableChunks)
+		{
+			if (m_UnusedSaveCountDown > 0)
+			{
+				--m_UnusedSaveCountDown;
+				// Wait and don't save yet, in hopes that other worlds managed to free enough.
+				// We'll get back here next iteration
+			}
+			else
+			{
+				// We have waited for 5 iterations (10 seconds), and we still can't free enough,
+				// and other worlds didn't free enough. Let's resort to saving unused dirty chunks.
+
+				// Make sure we haven't saved recently.
+				if (m_WorldAge - m_LastUnusedSave > std::chrono::seconds(20))
```

Comment:
@worktycho @tigerw I think I can get rid of this delay, along with `if (m_TickRand.randInt(3) == 0)`, without introducing LRU (see LRU disadvantages in that other comment). 

Here's how:

- Rather than cWorld, It's cRoot that calculates `UnloadChance`.
- Worlds simply fetch `UnloadChance` from cRoot.
- cRoot tracks which worlds fetched the current unloadChance via a simple bool array.
- Once all worlds fetch the current UnloadChance, cWorld calculates a new one.

This has the added advantage of making unloading truly uniform across worlds.